cloud/src/autogluon/cloud/utils/iam.py

```diff
@@ -0,0 +1,295 @@
+                "sagemaker:CreateTransformJob",
+                "sagemaker:DeleteEndpoint",
+                "sagemaker:DeleteEndpointConfig",
+                "sagemaker:DeleteModel",
+                "sagemaker:UpdateArtifact",
+                "sagemaker:UpdateEndpoint",
+                "sagemaker:InvokeEndpoint"
+            ],
+            "Resource": [
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:artifact/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:transform-job/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:endpoint/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:training-job/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:model/*",
+                f"arn:aws:sagemaker:*:{POLICY_ACCOUNT_PLACE_HOLDER}:endpoint-config/*"
+            ]
+        },
+        {
+            "Sid": "IAM",
+            "Effect": "Allow",
+            "Action": [
+                "iam:PassRole"
+            ],
+            "Resource": [
+                f"arn:aws:iam::{POLICY_ACCOUNT_PLACE_HOLDER}:role/*",
+            ]
+        },
+        {
+            "Sid": "CloudWatchDescribe",
+            "Effect": "Allow",
+            "Action": [
+                "logs:DescribeLogStreams"
+            ],
+            "Resource": [
+                f"arn:aws:logs:*:{POLICY_ACCOUNT_PLACE_HOLDER}:log-group:*"
+            ]
+        },
+        {
+            "Sid": "CloudWatchGet",
+            "Effect": "Allow",
+            "Action": [
+                "logs:GetLogEvents"
+            ],
+            "Resource": [
+                f"arn:aws:logs:*:{POLICY_ACCOUNT_PLACE_HOLDER}:log-group:*:log-stream:*"
+            ]
+        },
+        {
+            "Sid": "S3Object",
+            "Effect": "Allow",
+            "Action": [
+                "s3:PutObject",
+                "s3:PutObjectAcl",
+                "s3:GetObject",
+                "s3:GetObjectAcl",
+                "s3:AbortMultipartUpload"
+            ],
+            "Resource": [
+                f"arn:aws:s3:::{POLICY_BUCKET_PLACE_HOLDER}/*",
+                f"arn:aws:s3:::{POLICY_BUCKET_PLACE_HOLDER}",
+                "arn:aws:s3:::*SageMaker*",
+                "arn:aws:s3:::*Sagemaker*",
+                "arn:aws:s3:::*sagemaker*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetBucketAcl",
+                "s3:PutObjectAcl"
+            ],
+            "Resource": [
+                f"arn:aws:s3:::{POLICY_BUCKET_PLACE_HOLDER}/*",
+                f"arn:aws:s3:::{POLICY_BUCKET_PLACE_HOLDER}",
+                "arn:aws:s3:::*SageMaker*",
+                "arn:aws:s3:::*Sagemaker*",
+                "arn:aws:s3:::*sagemaker*"
+            ]
+        },
+        {
+            "Sid": "S3Bucket",
+            "Effect": "Allow",
+            "Action": [
+                "s3:CreateBucket",
```

Comment:
Do we really need `CreateBucket`? What is it for?

Reply:
sagemaker create buckets to store some intermediate artifacts

Reply:
Just curious what if the bucket already exists, what do we do? S3 bucket names are global so it is highly likely

Reply:
Then it's a sagemaker problem I guess. But sagemaker will create a bucket with a format like `sagemaker-region-account_id`